### DISCUSSION TITLE
Even after using CustomScreenshot function that's given in the gauge documentation we are unable to see the screenshots in the html reports.

### DISCUSSION THREAD

**Discussion by dhanushng88** [getgauge/gauge#2120] [General]
Code: 
gauge.customScreenshotWriter = async function () {
    const screenshotFilePath = path.join(process.env['gauge_screenshots_dir'], `screenshot-${process.hrtime.bigint()}.png`);
    await screenshot({ path: screenshotFilePath });
    return path.basename(screenshotFilePath);
};

afterStep(async () => {
    if (process.env.api_test_run === 'false') {
        gauge.screenshot();
    }
});

Gauge version: 1.2.1
Commit Hash: fa7918b

Plugins
-------
html-report (4.0.12)
js (2.3.17)
json-report (0.3.6)
screenshot (0.1.0)
xml-report (0.2.
[NoScreenShotsPresent.pdf](https://github.com/getgauge/gauge/files/7283355/NoScreenShotsPresent.pdf)
3)

**Comment by dhanushng88** [comment]
@zabil @NivedhaSenthil @negiDharmendra Can you please assist me regarding the Above issue?

**Comment by zabil** [comment]
This feature is only available only if you upgrade Gauge to version to 1.4.1 and html plugin version to 4.1.2. Please try upgrading it.

**Reply by dhanushng88** [reply]
@zabil Yes I have upgraded the gauge and Html-version as mentioned by you.. But still Unable to see the screenshots in Html reports while Running though Jenkins. This works fine in local runs from VS code using Run spec and after the execution we can see the screenshots, But the same when we run using Jenkins Jobs we are unable to see the screenshots.

Please advice me to proceed further

we have this in default.properties file

#The path to the gauge reports directory. Should be either relative to the project directory or an absolute path
gauge_reports_dir = reports

#Set as false if gauge reports should not be overwritten on each execution. A new time-stamped directory will be created on each execution.
overwrite_reports = true

# Set to false to disable screenshots on failure in reports.
screenshot_on_failure = true

gauge.customScreenshotWriter = async function () {
    const screenshotFilePath = path.join(process.env['gauge_screenshots_dir'], `screenshot-${process.hrtime.bigint()}.png`);
    await screenshot({ path: screenshotFilePath });
    return path.basename(screenshotFilePath);
};

afterStep(async () => {
    if (process.env.api_test_run === 'false') {
        gauge.screenshot();
    }
});

Gauge version: 1.4.1
Commit Hash: a82478a

Plugins
-------
html-report (4.1.2)
js (2.3.17)
json-report (0.3.6)
screenshot (0.1.0)
xml-report (0.2.3)

**Reply by zabil** [reply]
There's extra configuration required to view screenshots on jenkins please refer
https://github.com/getgauge/html-report/issues/219
